### PR TITLE
PUBDEV-8345: make H2O frame out of anova glm model summary

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes task in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8345.

I have completed:
1. backend support to convert model summary to H2OFrame
2. java unit test
3. Python api and Python unit test
4.R api and R unit test

@seb:  is this what you are looking for?  Feel free to change names and stuff.

### CODE REVIEW

**Review by sebhrusen — COMMENTED** [review]
Thanks for those quick changes @wendycwong 

As the new method added here to obtain the result frame is supposed to be on par with `predict` and `transform`, I just want to be sure we all agree on the name.

Reminder:
Python:
```python
model.predict(fr, **kwargs) -> H2OFrame    # most models

model.transform(fr) -> H2OFrame              # TargetEncoder, PCA, SVD
```
R:
```
h2o.predict(model, fr, ...)

h2o.transform(model, fr, ...)
```
Now the new method will be first used by ANOVA and MaxR2, and could be expected to be implemented for any model that can't predict or transform but provides a definitive result.
Some suggestions:
```python
model.result() -> H2OFrame

model.result_frame() -> H2OFrame

model.as_frame() -> H2OFrame
```
/
```
h2o.result(model)
h2o.result_frame(model)
as.h2o(model)
```

I tend to prefer the simpler form `result()`.
wdyt? @wendycwong @michalkurka @ledell

File: h2o-algos/src/main/java/hex/schemas/ANOVAGLMModelV3.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
     @API(help="AnovaGLM transformed predictor frame key.  For debugging purposes only")
     String transformed_columns_key;
 
+    @API(help="ANOVA table frame key containing same results in the model summary.")
+    String result_frame_key;
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/schemas/ANOVAGLMModelV3.java:22]: I need it to be string in order to pass to R/python clients.  The FrameKey is kept for Java clients and also to untrack the key at the end of model building.

File: h2o-algos/src/main/java/hex/anovaglm/ANOVAGLMModel.java [hunk 2]
```diff
@@ -162,12 +175,17 @@ public ANOVAGLMModelOutput(ANOVAGLM b, DataInfo dinfo) {
    * @return
    */
   public TwoDimTable generateSummary(String[] modelNames, GLMModel[] glmModels, int[] degreeOfFreedom){
-    String[] names = new String[]{"Model", "family", "link", "SS", "DF", "MS", "F", "p-value"};
+    String[] names = new String[]{"model", "family", "link", "ss", "df", "ms", "f", "p_value"};
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/anovaglm/ANOVAGLMModel.java]: Yes, it should be predictors or predictors interaction

File: h2o-algos/src/main/java/hex/schemas/ANOVAGLMModelV3.java [hunk 3]
```diff
@@ -18,6 +18,9 @@
     @API(help="AnovaGLM transformed predictor frame key.  For debugging purposes only")
     String transformed_columns_key;
 
+    @API(help="ANOVA table frame key containing same results in the model summary.")
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-algos/src/main/java/hex/schemas/ANOVAGLMModelV3.java]: Yes, I changed the description.

**Comment by wendycwong** [comment]
@seb:  I changed the name to result.  Just need to add the doc description for it.

**Review by sebhrusen — APPROVED** [review]
looks good to me, as soon as everyone is fine with the new method!
Thanks @wendycwong

**Pull request merged**